google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -1082,13 +1083,16 @@ public TableDataList call() {
                           completeTableId.getProject(),
                           completeTableId.getDataset(),
                           completeTableId.getTable(),
-                          optionsMap);
+                          updatedOptionsMap);
                 }
               },
               serviceOptions.getRetrySettings(),
               EXCEPTION_HANDLER,
               serviceOptions.getClock());
       String cursor = result.getPageToken();
+      if (!Strings.isNullOrEmpty(cursor)) {
+        optionsMap = optionMap(TableDataListOption.startIndex(0));
+      }
       return Tuple.of(
           new PageImpl<>(
               new TableDataPageFetcher(tableId, schema, serviceOptions, cursor, optionsMap),
```

Comment:
It would be more clear if you keep `optionalMap` variable final and declare:

```final Map<BigQueryRpc.Option, ?> updatedOptionsMap = Strings.isNullOrEmpty(cursor) ? optionsMap : optionMap(TableDataListOption.startIndex(0))```

Reply:
If i initialized updateOptionmap  before making a rpc call so it's does not effect current bug because every time it's initialized with same value so ternary with initialization is not possible.

Reply:
You need to initialize `updateOptionmap` after `String cursor = result.getPageToken();`

Reply:
if i initialize ```updateOptionmap``` before after ```String cursor``` then it's value does not available to second rpc call 
and index does not update for second call that's why it's require to initialize it's before making an rpc call.